runtime/jcl/common/criusupport.c

```diff
@@ -44,3 +54,94 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
 
 	return res;
 }
+
+jint JNICALL
+Java_com_ibm_jvm_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
+							jclass unused,
+							jobject checkPointDir,
+							jboolean keepRunning,
+							jboolean shellJob,
+							jboolean extUnixSupport,
+							jint logLevel,
+							jobject logFile)
+{
+	jint rc = UNSUPPORTED_OPERATION;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	char directoryBuf[STRING_BUFFER_SIZE] = {0};
+	char *directoryUTF = NULL;
+	char logFileBuf[STRING_BUFFER_SIZE] = {0};
+	char *logFileUTF = NULL;
+
+	if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_CRIU_SUPPORT)) {
+#if defined(LINUX)
+		j9object_t cpDir = NULL;
+		j9object_t log = NULL;
+		IDATA dirFD = 0;
+		PORT_ACCESS_FROM_VMC(currentThread);
+
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+
+		cpDir = J9_JNI_UNWRAP_REFERENCE(checkPointDir);
+		log = J9_JNI_UNWRAP_REFERENCE(logFile);
+
+		directoryUTF = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, cpDir, J9_STR_NULL_TERMINATE_RESULT, "", 0, directoryBuf, STRING_BUFFER_SIZE, NULL);
+		if (NULL == directoryUTF) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto release;
+		}
+
+		logFileUTF = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, log, J9_STR_NULL_TERMINATE_RESULT, "", 0, logFileBuf, STRING_BUFFER_SIZE, NULL);
+		if (NULL == logFileUTF) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto freeDir;
+		}
+
+		dirFD = open(directoryUTF, O_DIRECTORY);
+		if (dirFD < 0) {
+			rc = INVALID_ARGUEMENTS;
+			goto freeLog;
+		}
+
+		criu_set_images_dir_fd(dirFD);
+		criu_set_shell_job(shellJob);
+		criu_set_log_level(logLevel);
+		criu_set_log_file(logFileUTF);
+		criu_set_leave_running(keepRunning);
+		criu_set_ext_unix_sk(extUnixSupport);
+
+		if (FALSE == vmFuncs->jvmCheckpointHooks(currentThread)) {
+			rc = JVM_CHECKPOINT_FAILURE;
+			goto freeLog;
+		}
+
+		if (0 > criu_dump()) {
```

Comment:
```suggestion
		if (criu_dump() != 0) {
```

Based on the limited documentation at [1], I think any non-0 result is an error.  Also, given the LHS isn't a variable, you don't need to invert it.  Much easier to read if it's not inverted.

[1] https://criu.org/C_API

Reply:
On the restore side a positive value is returned.

Reply:
<img width="1085" alt="image" src="https://user-images.githubusercontent.com/12551021/122237700-8874d200-ce8d-11eb-973e-73d36ba6fa74.png">

Reply:
Have you confirmed that?  I would have expected the `>0` to apply to an explicit call to `criu_restore`

Reply:
yes if you restore with criu CLI (eg. `criu restore -D ./cpData --shell-job`) the result is > 0